drivers/hmis/lib/hmis_data_cleanup/util.rb

```diff
@@ -27,43 +37,87 @@ def self.clear_enrollment_export_ids!
     # we are dealing with a single-CoC installation at the time of writing this.
     def self.update_all_enrollment_cocs!(coc_code)
       without_papertrail_or_timestamps do
-        Hmis::Hud::Enrollment.hmis.update_all(EnrollmentCoC: coc_code)
+        rows_affected = Hmis::Hud::Enrollment.hmis.where.not(EnrollmentCoC: coc_code).update_all(EnrollmentCoC: coc_code)
+        Rails.logger.info "Updated EnrollmentCoC on #{rows_affected} enrollments"
       end
     end
 
     # Assign Household ID where missing
-    #
-    # Note: If this gets to be a large number, an upsert is probably worth doing.
-    # We could `scope.find_in_batches` and then enrollment.assign_attributes and finally "import" the batch with a conflict key of id.
     def self.assign_missing_household_ids!
-      scope = Hmis::Hud::Enrollment.hmis.where(household_id: nil)
-      Rails.logger.info "Assigning household id to #{scope.size} enrollments"
-      scope.each do |enrollment|
-        hh_id = Hmis::Hud::Base.generate_uuid
-        without_papertrail_or_timestamps do
-          enrollment.update_columns(household_id: hh_id) # skips callbacks
-        end
+      Hmis::Hud::Enrollment.hmis.where(household_id: [nil, '']).find_in_batches do |batch|
+        batch.each { |enrollment| enrollment.HouseholdID = Hmis::Hud::Base.generate_uuid }
```

Comment:
This is fine, but if you want it to be consistent over time, something like https://github.com/greenriver/hmis-warehouse/blob/30d691487c2fae4f56815034f24594a66fa3357a/drivers/hmis_csv_importer/app/models/hmis_csv_importer/hmis_csv_cleanup/fix_blank_household_ids.rb#L14 might be better.

Reply:
updated thanks!